src/test/java/com/learning/pages/swaglabs/SwagLabsLoginPageTest.java

```diff
@@ -39,49 +40,99 @@ void shouldErrorMessageBeVisibleWhenUnsuccessfulLogin(String username, String pa
     //When
     swagLabsLoginPage.open();
     swagLabsLoginPage.performLogin(username, password);
-    String actualErrorMessage = swagLabsLoginPage.getErrorText();
     //Then
+    String actualErrorMessage = swagLabsLoginPage.getErrorText();
     assertEquals(actualErrorMessage, expectedErrorMessage);
   }
 
-  @Test(dataProvider = "unsuccessfulLoginData")
-  void shouldErrorMessageBeVisibleWhenLackOfPassword(String username, String password) {
+  @Test
+  void shouldErrorMessageBeVisibleWhenLackOfPassword() {
     //Given
+    String username = "standard_user";
     String expectedErrorMessage = "Epic sadface: Password is required";
     //When
     swagLabsLoginPage.open();
     swagLabsLoginPage.performLogin(username, "");
-    String actualErrorMessage = swagLabsLoginPage.getErrorText();
     //Then
+    String actualErrorMessage = swagLabsLoginPage.getErrorText();
     assertEquals(actualErrorMessage, expectedErrorMessage);
   }
 
-  @Test(dataProvider = "unsuccessfulLoginData")
-  void shouldErrorMessageBeVisibleWhenLackOfUsername(String username, String password) {
+  @Test
+  void shouldErrorMessageBeVisibleWhenLackOfUsername() {
     //Given
+    String password = "secret_sauce";
     String expectedErrorMessage = "Epic sadface: Username is required";
     //When
     swagLabsLoginPage.open();
     swagLabsLoginPage.performLogin("", password);
-    String actualErrorMessage = swagLabsLoginPage.getErrorText();
     //Then
+    String actualErrorMessage = swagLabsLoginPage.getErrorText();
     assertEquals(actualErrorMessage, expectedErrorMessage);
   }
 
+  @Test
+  void shouldFailToLoginWhenLackOfCredentials() {
+    //Given
+    String expectedURL = swagLabsLoginPage.getPageURL();
+    //When
+    swagLabsLoginPage.open();
+    swagLabsLoginPage.performLogin("", "");
+    //Then
+    assertEquals(browser.getCurrentPageURL(), expectedURL);
+  }
+
+  @Test
+  void shouldNotGoToInventoryPageWithoutLogin() {
+    //Given
+    String expectedURL = swagLabsLoginPage.getPageURL();
+    //When
+    swagLabsInventoryPage.open();
+    //Then
+    assertEquals(browser.getCurrentPageURL(), expectedURL);
+  }
+
+  @Test(dataProvider = "lockedUserData")
+  void shouldNotLoginWhenLockedUser(String username, String password) {
+    //Given
+    String expectedErrorMessage = "Epic sadface: Sorry, this user has been locked out.";
+    //When
+    swagLabsLoginPage.open();
+    swagLabsLoginPage.performLogin(username, password);
+    //Then
+    assertEquals(swagLabsLoginPage.getErrorText(), expectedErrorMessage);
+  }
+
+  @Test
+  void shouldFailToLoadPageWithProblemUserLogin() {
+    //Given
+    String username = "problem_user";
+    String password = "secret_sauce";
+    //When
+    swagLabsLoginPage.open();
+    swagLabsLoginPage.performLogin(username, password);
+    assertThrows(() -> browser.waitForPageToLoad(swagLabsInventoryPage));
```

Comment:
This one fails due to `waitForPageToLoad`. 
It does not detects that all images on website were not loaded. 
When I add `wait.until(ExpectedConditions.visibilityOfAllElements(elements))` then google page fails to load and this one still not throwing error. Any suggestions how to do it? Or this is to catch only with manual testing?

Reply:
I'd have to take a deeper look into this.
However, I would not decide if page was loaded based on images :/ more "crucial" page elements should be chosen. Unless test is to explicitly verify that given image was loaded, I would not use images for anything. To say more... in my current project we don't check images even in suite we run during UI release pipeline.

Reply:
I think in this case (shop page) is important to have pictures of products on right place. 
So maybe as you said in this particular test is better to check explicitly every picture from offering cards rather then visibility of whole offering card element.